### ISSUE TITLE
Bind does not properly validate data input

### ISSUE THREAD

**Issue by TomKaltz** [Rotorsoft/eventually-monorepo#52]
bind() does not properly validate data passed in based on the name parameter.  Currently any data shape that matches the union of all Message types are accepted. Is there a way at compile time to validate whether we are passing the correct data for a particular Message?

**Comment by Rotorsoft** [comment]
I think so. Will push a fix soon, something like this:

```
/**
 * Binds message arguments
 * @param name the message name
 * @param data the message payload
 * @param target optional command target args
 * @returns The bound message
 */
export const bind = <M extends Messages, N extends keyof M>(
  name: N,
  data: Readonly<M[N]>,
  target?: CommandTarget
): Message<M> | Command<M> =>
  ({ name, data, ...target }) as Message<M> | Command<M>;
```

**Comment by TomKaltz** [comment]
Closed by a38a4e5aea12500e61fcfb046fa22fd7229fc26d
